### PR TITLE
wire replicator transmission to domain APIs

### PR DESCRIPTION
**Pull request by wxing1292**
solve #569

### CODE REVIEW

File: service/frontend/handler.go [hunk 1]
```diff
@@ -117,6 +124,8 @@ func NewWorkflowHandler(
 		hSerializerFactory: persistence.NewHistorySerializerFactory(),
 		domainCache:        cache.NewDomainCache(metadataMgr, sVice.GetLogger()),
 		rateLimiter:        common.NewTokenBucket(config.RPS, common.NewRealTimeSource()),
+		kafkaProducer:      kafkaProducer,
```
Comment (samarabbas) [inline service/frontend/handler.go]: Let's not use any Kafka naming at this layer.

File: service/worker/domainReplicationTaskHandler.go [hunk 2]
```diff
@@ -65,8 +67,45 @@ func NewDomainReplicator(metadataManager persistence.MetadataManager, logger bar
 	}
 }
 
-// handleDomainReplicationTask handle the domain replication task
-func (domainReplicator *domainReplicatorImpl) HandleReceiveTask(task *replicator.DomainTaskAttributes) error {
+// HandleTransmissionTask handle transmission of the domain replication task
+func (domainReplicator *domainReplicatorImpl) HandleTransmissionTask(kafka messaging.Producer, domainOperation replicator.DomainOperation,
```
Comment (samarabbas) [inline service/worker/domainReplicationTaskHandler.go]: Let's move this logic into a separate component which lives within frontend.

Comment (wxing1292) [reply service/worker/domainReplicationTaskHandler.go]: done

File: service/frontend/handler.go (hunk 1 above)
Comment (wxing1292) [reply service/frontend/handler.go]: reduced to just the domainReplicator dependency

**Pull request merged**